File: server/app/schemas/room_speech.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class RoomSpeechCreateRequest(BaseModel):
    content: str


class RoomSpeechItem(BaseModel):
    user_id: uuid.UUID
    nickname: str
    content: str
    created_at: datetime
    expires_at: datetime

    model_config = {"from_attributes": True}


class RoomSpeechSubmitResult(BaseModel):
    content: str
    created_at: datetime
    expires_at: datetime


class RoomSpeechDeleteResult(BaseModel):
    ok: bool = True

File: app/lib/features/challenge_space/models/room_speech.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'room_speech.freezed.dart';
part 'room_speech.g.dart';

@freezed
class RoomSpeech with _$RoomSpeech {
  const factory RoomSpeech({
    @JsonKey(name: 'user_id') required String userId,
    required String nickname,
    required String content,
    @JsonKey(name: 'created_at') required DateTime createdAt,
    @JsonKey(name: 'expires_at') required DateTime expiresAt,
  }) = _RoomSpeech;

  factory RoomSpeech.fromJson(Map<String, dynamic> json) =>
      _$RoomSpeechFromJson(json);
}
